[Feature] Add group selector logic

Dysproz opened the pull request:
In this review:
* I added logic for selecting only specific groups from inventory. It works with multiple groups and also with standard indices like: [3], [:3], [3:]. Groups may be specified with `-g` or `--groups` flag So it's possible to connect for example: Only to first 3 hosts from groupA and last host from groupB. (`-g groupA[:3] groupB[-1]). Old logic works the same way - it's still possible to pass only inventory and expect to open all hosts.
* I added unittests for inventoryparser.py and 2 functions that I added to main.py
* I added sample inventory file (under *tests/files/*) - don't worry, I made sure that there's no confidential information inside :) Only mock data.

P.S.
I checked my code with *flake8* with --max-line-length parameter set to 100.

Dysproz committed:
Add group limiting logic

psykulsk commented on ansibleinviewer/inventoryparser.py:
```diff
@@ -68,4 +105,11 @@ def parse_inventory_data_dict(self, inventory_data: dict) -> Dict:
+                    host_list.append(host)
+            return host_list
+        else:
+            return list(self.hosts)
```
I think we should pass the group as an argument to the get_hosts method instead of the __init__.

The groups slicing is cool, but I do not think it will be that useful for the user. He will probably not know which hosts are the first 4 hosts, or the last 3 in the inventory. I was thinking about selecting the whole group as one way of selecting hosts and providing a list of hostnames to show as the other.

Dysproz replied:
Done

psykulsk commented on ansibleinviewer/main.py:
```diff
@@ -40,12 +40,58 @@ def create_tmux_script(hosts: List[Host], vertical_panes) -> str:
+            parsed_group = (group, None)
+        parsed_groups.append(parsed_group)
+    return parsed_groups
+
```
As mentioned above, I think user should be able to select only the whole group as he most probably won't know which slice of the group has the hosts he wants.

Dysproz replied:
In new patchset user can now select only whole groups and select groups that should be omitted

psykulsk commented on ansibleinviewer/tests/requirements.txt:
```diff
@@ -0,0 +1,3 @@
+parameterized==0.7.0
+pytest==5.2.2
```
In case of this module I would make the requirements as loose as possible as the module will be probably installed together with other python modules with their requirements. Forcing concrete versions would force the user to use a separate venv just for this module.

Dysproz replied:
I added requirements up to second version tag.

psykulsk commented on README.md:
```diff
@@ -19,10 +19,16 @@ python3 -m pip install .
+#### Possible flags
+
+* `-i`, `--inventory` - Path to ansible inventory
+* `-g`, `--groups` - Inventory groups of hosts to connect to
```
I would like groups to option to be used like this: -g group1,group2,group3

Dysproz replied:
It's already possible. In new patchset I update it only to utilize : ans connector, because ! is bash restricted character

psykulsk reviewed (changes requested):
Thanks for the PR :)! I like the code, but I was thinking of doing the group selection a little bit differently. I don't the group slicing option is going to be useful for the user. I think the whole group or multiple groups should be always selected f.e. like this: ansibleinviewer -i inventory.yml -g group1,group2.

Dysproz committed:
Update group selector to comments

Dysproz commented:
Hi @psykulsk,
Thank you for quick review. 
I uploaded new patchset where I refactored group selection.
I dumped slicing. However, I added possibility to deselect certain groups with *!*.
Unfortunately, it was not possible to do that with nargs (`group1 group2 !group3`), because *!* is bash restricted character.
So multiple groups may be specified with *:* character as connector (ansible utilizes similar notation)
e.g. `group1:group2:!group3`.

Dysproz committed:
Fix flake8 style

Dysproz committed:
Bring bask dist predefined type ehh ehh

psykulsk commented on ansibleinviewer/main.py:
```diff
@@ -40,12 +40,36 @@ def create_tmux_script(hosts: List[Host], vertical_panes) -> str:
+                * groups that should be ommited
+    :rtype: list
+    """
+    provided_groups = args_groups.split(':')
```
I think you need to check whether args_groups is not None

psykulsk replied:
Also, a unit test for it would be good.

psykulsk commented on ansibleinviewer/tests/test_inventoryparser.py:
```diff
@@ -0,0 +1,69 @@
+
+    def setUp(self):
+        dir_path = os.path.dirname(os.path.realpath(__file__))
+        inventory_data = self.load_inventory_file(f'{ dir_path}/files/inventory.yml')
```
The inventory.yml file was not pushed as I've added `inventory.yml` to the .gitignore. You can remove it from the .gitignore or rename the test file.

Dysproz committed:
Add None handling to parse_inventory_groups, remove inventory from
gitignore

Dysproz commented:
Oh! I haven't even noticed that inventory.yml was ignored.

psykulsk reviewed (approved):
LGTM

Dysproz commented:
:dancer:
